pkg/cmd/roachtest/spec/machine_type.go

```diff
@@ -16,26 +16,51 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachprod/vm"
 )
 
-// AWSMachineType selects a machine type given the desired number of CPUs and
-// memory per CPU ratio. Also returns the architecture of the selected machine type.
-func AWSMachineType(cpus int, mem MemPerCPU, arch vm.CPUArch) (string, vm.CPUArch) {
-	// TODO(erikgrinaker): These have significantly less RAM than
-	// their GCE counterparts. Consider harmonizing them.
-	family := "c6id" // 2 GB RAM per CPU
+// AWSMachineType selects a machine type given the desired number of CPUs, memory per CPU,
+// support for locally-attached SSDs and CPU architecture. It returns a compatible machine type and its architecture.
+//
+// When MemPerCPU is Standard, the memory per CPU ratio is 4 GB. For High, it is 8 GB.
+// For Auto, it's 4 GB up to and including 16 CPUs, then 2 GB. Low is not supported.
+//
+// N.B. in some cases, the selected architecture and machine type may be different from the requested one. E.g.,
+// graviton3 with >= 24xlarge (96 vCPUs) isn't available, so we fall back to (c|m|r)6i.24xlarge.
+// N.B. cpus is expected to be an even number; validation is deferred to a specific cloud provider.
+//
+// At the time of writing, the intel machines are all third-generation Xeon, "Ice Lake" which are isomorphic to
+// GCE's n2-(standard|highmem|custom) _with_ --minimum-cpu-platform="Intel Ice Lake" (roachprod's default).
+func AWSMachineType(
+	cpus int, mem MemPerCPU, shouldSupportLocalSSD bool, arch vm.CPUArch,
+) (string, vm.CPUArch) {
+	family := "m6i" // 4 GB RAM per CPU
 	selectedArch := vm.ArchAMD64
+
 	if arch == vm.ArchFIPS {
+		// N.B. FIPS is available in any AMD64 machine configuration.
 		selectedArch = vm.ArchFIPS
 	} else if arch == vm.ArchARM64 {
-		family = "c7g" // 2 GB RAM per CPU (graviton3)
+		family = "m7g" // 4 GB RAM per CPU (graviton3)
 		selectedArch = vm.ArchARM64
 	}
 
-	if mem == High {
-		family = "m6i" // 4 GB RAM per CPU
+	switch mem {
+	case Auto:
+		if cpus > 16 {
+			family = "c6i" // 2 GB RAM per CPU
+
+			if arch == vm.ArchARM64 {
+				family = "c7g" // 2 GB RAM per CPU (graviton3)
+				selectedArch = vm.ArchARM64
```

Comment:
Nit: technically not needed, since this is already taken care of in the code above the switch.